cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/hbase/CombinedMetricsScanner.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.data2.dataset2.lib.table.hbase;
+
+import co.cask.cdap.api.common.Bytes;
+import co.cask.cdap.api.dataset.table.Result;
+import co.cask.cdap.api.dataset.table.Row;
+import co.cask.cdap.api.dataset.table.Scanner;
+import com.google.common.annotations.VisibleForTesting;
+
+import java.util.Iterator;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.TreeMap;
+import javax.annotation.Nullable;
+
+/**
+ * Implements Scanner on merge data from v2 and v3 metrics table.
+ */
+public class CombinedMetricsScanner implements Scanner {
+
+  private final Scanner v2TableScanner;
+  private final Scanner v3TableScanner;
+  private Row v2TableNextRow;
+  private Row v3TableNextRow;
+
+  @VisibleForTesting
+  public CombinedMetricsScanner(Scanner v2TableScanner, Scanner v3TableScanner) {
+    this.v3TableScanner = v3TableScanner;
+    this.v2TableScanner = v2TableScanner;
+    this.v3TableNextRow = v3TableScanner.next();
+    this.v2TableNextRow = v2TableScanner.next();
+  }
+
+  @Nullable
+  @Override
+  public Row next() {
+    Row resultRow;
+    // both the scanners are exhausted, so return null
+    if (v2TableNextRow == null && v3TableNextRow == null) {
+      return null;
+    }
+
+    // v2 scanner is exhausted, so return row from v3 table
+    if (v2TableNextRow == null) {
+      return getV3Row();
```

Comment:
the name of the method (getV3Row) suggests that it only gets. But it also advances the v3 table scanner and replaces the nextV3Row with the next result. It would be better to name the method something like advanceV3Row or advanceV3Scanner.